src/server/bitops_family.cc

```diff
@@ -0,0 +1,285 @@
+// Helper functions to access the data or change it
+
+void UpdateBitMapValue(DbIndex index, DbSlice& db_slice, PrimeIterator& it, std::string_view key,
+                       std::string_view value, bool added) {
+  db_slice.PreUpdate(index, it);
+  it->second.SetString(value);
+  db_slice.PostUpdate(index, it, key, !added);
+}
+
+class OverRideValue {
+  EngineShard* shard_ = nullptr;
+  DbIndex index_ = 0;
+
+ public:
+  explicit OverRideValue(const OpArgs& args) : shard_{args.shard}, index_{args.db_ind} {
+  }
+
+  OpResult<bool> Set(std::string_view key, int32_t offset, bool bit_value);
+};
+
+OpResult<bool> OverRideValue::Set(std::string_view key, int32_t offset, bool bit_value) {
+  auto& db_slice = shard_->db_slice();
+
+  DCHECK_LT(index_, db_slice.db_array_size());
+  DCHECK(db_slice.IsDbValid(index_));
+
+  std::pair<PrimeIterator, bool> add_res;
+  try {
+    add_res = db_slice.AddOrFind(index_, key);
+  } catch (const std::bad_alloc&) {
+    return OpStatus::OUT_OF_MEMORY;
+  }
+  bool old_value = false;
+  PrimeIterator& it = add_res.first;
+  if (add_res.second) {  // this is a new entry in the "table"
+    std::string new_entry(GetByteIndex(offset) + 1, 0);
+    old_value = SetBitValue(new_entry, offset, bit_value);
+    UpdateBitMapValue(index_, db_slice, it, key, new_entry, add_res.second);
+  } else {
+    if (it->second.ObjType() != OBJ_STRING) {
+      return OpStatus::WRONG_TYPE;
+    }
+    bool reset = false;
+    std::string existing_entry{GetString(shard_, it->second)};
+    if ((static_cast<int32_t>(existing_entry.size()) * OFFSET_FACTOR) <=
+        offset) {  // need to resize first
+      existing_entry.resize(GetByteIndex(offset) + 1, 0);
+      reset = true;
+    }
+    old_value = SetBitValue(existing_entry, offset, bit_value);
+    if (reset || old_value != bit_value) {  // we made a "real" change to the entry, save it
+      UpdateBitMapValue(index_, db_slice, it, key, existing_entry, add_res.second);
+    }
+  }
+  return old_value;
+}
+// ------------------------------------------------------------------------- //
+//  Impl for the command functions
+void BitPos(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitCount(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitField(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitFieldRo(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendLong(0);
+}
+
+void BitOp(CmdArgList args, ConnectionContext* cntx) {
+  (*cntx)->SendOk();
+}
+
+void GetBit(CmdArgList args, ConnectionContext* cntx) {
+  // Support for the command "GETBIT key offset"
+  // see https://redis.io/commands/getbit/
+
+  int32_t offset{0};
+  std::string_view key = ArgS(args, 1);
+
+  if (!absl::SimpleAtoi(ArgS(args, 2), &offset)) {
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+  auto cb = [&](Transaction* t, EngineShard* shard) { return ReadValue(t->GetOpArgs(shard), key); };
```

Comment:
I suggest that instead of returning the whole string back to coordinator, you will figure out the bit in the shard thread, similarly to how you did with SetBit. And then just return `OpResult<bool>`.

Reply:
ok, this was done as the first "version" I can re-right this case.